Modules/_winapi.c

```diff
@@ -774,12 +774,139 @@ gethandle(PyObject* obj, const char* name)
     return ret;
 }
 
+static PyObject*
+sortenvironmentkey(PyObject *module, PyObject *item)
+{
+    return _winapi_LCMapStringEx_impl(NULL, LOCALE_NAME_INVARIANT, LCMAP_UPPERCASE, item);
+}
+
+static PyMethodDef sortenvironmentkey_def = {
+    "sortenvironmentkey",
+    _PyCFunction_CAST(sortenvironmentkey),
+    METH_O,
+    "",
+};
+
+static PyObject *
+normalize_environment(PyObject* environment)
+{
+    PyObject *result = NULL, *keys = NULL, *keyfunc = NULL, *kwnames = NULL;
+
+    keys = PyMapping_Keys(environment);
+    if (keys == NULL) {
+        return NULL;
+    }
+
+    keyfunc = PyCFunction_New(&sortenvironmentkey_def, NULL);
+    if (keyfunc == NULL) {
+        goto error;
+    }
+    kwnames = PyTuple_New(1);
+    if (kwnames == NULL) {
+        goto error;
+    }
+    assert(PyTuple_SetItem(kwnames, 0, PyUnicode_FromString("key")) == 0);
```

Comment:
```suggestion
    kwnames = Py_BuildValue("(s)", "key");
    if (kwnames == NULL) {
        goto error;
    }
```

Most importantly, this removes the expression in `assert()`, which would be skipped completely in a Release build and cause the call to break.

In general, we should never use assert to check the result of our own C API calls - only our own assumptions (e.g. the `assert(PyList_CheckExact(...))` case would be fine, because we want to know if/when the source code changes, but it has no runtime effect).